provider/spark.go

```diff
@@ -1401,11 +1401,56 @@ func (d *DatabricksExecutor) SparkSubmitArgs(destPath filestore.Filepath, cleanQ
 	credentialConfigs := store.CredentialsConfig()
 	argList = append(argList, credentialConfigs...)
 
-	argList = append(argList, "--source_list")
-	argList = append(argList, sourceList...)
+	// Databricks's API enforces a 10K-byte limit on job submit params, so to avoid a 400, we need to check to ensure
+	// the args are below this limit. If they exceed this limit, it's most likely due to the query and/or the list of
+	// sources, so we write these as a JSON file and read them from the PySpark runner script to side-step this constraint
+	if d.exceedsSubmitParamsTotalByteLimit(argList, cleanQuery, sourceList) {
+		if store.FilestoreType() != filestore.S3 {
+			return argList, fmt.Errorf("%s is not a currently support file store for writing submit params; supported types: %s", store.FilestoreType(), filestore.S3)
+		}
+		paramsFileId := uuid.New()
```

Comment:
Can we put this in a function so this outer function just deals with setting the args that are returned? A lot going on here